File: java/2145.java
class Solution {
    public int numberOfArrays(int[] d, int l, int u) {
        int n = d.length;
        long c = l, s = l, e = l;
        for (int i = 0; i < n; i++) {
            c += d[i];
            if (s > c)
                s = c;
            if (e < c)
                e = c;
        }
        return (int) Math.max(0, s - l + u - e + 1);
    }
}

File: javascript/2145.js
/**
 * @param {number[]} d
 * @param {number} l
 * @param {number} u
 * @return {number}
 */
var numberOfArrays = function (d, l, u) {
    let c = l, s = l, e = l;
    for (let i of d) {
        c += i;
        s = Math.min(s, c);
        e = Math.max(e, c);
        if (s - l + u - e + 1 < 0)
            return 0;
    }
    return s - l + u - e + 1;
};
